### PR TITLE
add confirmation before removing cell for #173481

### LINKED ISSUE #173481: Deleting cell should have confirmation before removing cell
Having just accidentally deleted a cell that had been running for over 4 hours, I'd like to suggest that there be a confirmation popup for running Jupyter cells to be deleted. Ideally there should be an option to confirm for *all* cells, but should be a default for cells that are currently executing.

### PR DESCRIPTION
**Pull request by Kalmaegi**
Fixes https://github.com/microsoft/vscode/issues/173481

### CODE REVIEW

File: src/vs/workbench/contrib/notebook/browser/controller/editActions.ts [hunk 1]
```diff
@@ -157,6 +160,37 @@ registerAction2(class DeleteCellAction extends NotebookCellAction {
...
+		const configService = accessor.get(IConfigurationService);
+
+		if (runState === NotebookCellExecutionState.Executing && configService.getValue(NotebookSetting.confirmDeleteRunningCell)) {
+			const dialogService = accessor.get(IDialogService);
+			const primaryButton = localize('confirmDeleteButton', 'Confirm Delete');
+
+			confirmation = await dialogService.confirm({
+				type: 'question',
+				message: localize('confirmDeleteButtonMessage', 'Are you sure you want to delete a running cell?'),
+				detail: localize('confirmDeleteButtonDetail', 'After deletion, the cell may not be recovered.'),
```
Comment (roblourens) [inline src/vs/workbench/contrib/notebook/browser/controller/editActions.ts:175]: That's not true, "undo" exists

File: src/vs/workbench/contrib/notebook/browser/controller/editActions.ts [hunk 2]
```diff
@@ -157,6 +160,37 @@ registerAction2(class DeleteCellAction extends NotebookCellAction {
...
+		const runState = notebookExecutionStateService.getCellExecution(context.cell.uri)?.state;
+		const configService = accessor.get(IConfigurationService);
+
+		if (runState === NotebookCellExecutionState.Executing && configService.getValue(NotebookSetting.confirmDeleteRunningCell)) {
+			const dialogService = accessor.get(IDialogService);
+			const primaryButton = localize('confirmDeleteButton', 'Confirm Delete');
+
+			confirmation = await dialogService.confirm({
+				type: 'question',
+				message: localize('confirmDeleteButtonMessage', 'Are you sure you want to delete a running cell?'),
```
Comment (roblourens) [inline src/vs/workbench/contrib/notebook/browser/controller/editActions.ts:174]: "This cell is running, are you sure you want to delete it?"

File: src/vs/workbench/contrib/notebook/browser/controller/editActions.ts [hunk 3]
```diff
@@ -157,6 +160,37 @@ registerAction2(class DeleteCellAction extends NotebookCellAction {
...
 		}
 
+		let confirmation: IConfirmationResult;
+		const notebookExecutionStateService = accessor.get(INotebookExecutionStateService);
+		const runState = notebookExecutionStateService.getCellExecution(context.cell.uri)?.state;
+		const configService = accessor.get(IConfigurationService);
+
+		if (runState === NotebookCellExecutionState.Executing && configService.getValue(NotebookSetting.confirmDeleteRunningCell)) {
+			const dialogService = accessor.get(IDialogService);
+			const primaryButton = localize('confirmDeleteButton', 'Confirm Delete');
```
Comment (roblourens) [inline src/vs/workbench/contrib/notebook/browser/controller/editActions.ts:170]: "Delete"

Comment (roblourens) [reply src/vs/workbench/contrib/notebook/browser/controller/editActions.ts:170]: And double-quotes on these strings

**Review by roblourens — COMMENTED** [review]
Thanks!

File: src/vs/workbench/contrib/notebook/browser/controller/editActions.ts (hunk 2 above)
Comment (weartist) [reply src/vs/workbench/contrib/notebook/browser/controller/editActions.ts:174]: Thanks! my english is not good haha

File: src/vs/workbench/contrib/notebook/browser/controller/editActions.ts (hunk 1 above)
Comment (weartist) [reply src/vs/workbench/contrib/notebook/browser/controller/editActions.ts:175]: I deleted a cell while it was running and undo didn't seem to take effect, so do we need to delete the description of the detail

Comment (roblourens) [reply src/vs/workbench/contrib/notebook/browser/controller/editActions.ts]: Works for me- that sounds like it would be a separate issue.

Comment (roblourens) [reply src/vs/workbench/contrib/notebook/browser/controller/editActions.ts]: You can delete the detail, and everything else here looks fine

**Review by roblourens — APPROVED** [review]

**Review by DonJayamanne — APPROVED** [review]

**Pull request merged**